rollup.config.js

```diff
@@ -1,37 +1,54 @@
-import commonjs from 'rollup-plugin-commonjs';
-import nodeResolve from 'rollup-plugin-node-resolve';
-import nodeBuiltins from 'rollup-plugin-node-builtins';
-import babel from 'rollup-plugin-babel';
-import banner from 'rollup-plugin-banner';
-import genHeader from './lib/header';
+import commonjs from 'rollup-plugin-commonjs'
+import resolve from 'rollup-plugin-node-resolve'
+import babel from 'rollup-plugin-babel'
+import typescript from 'rollup-plugin-typescript2'
+import banner from 'rollup-plugin-banner'
+import genHeader from './lib/header'
 
-export default [{
-	input: 'index.js',
-	output: {
-		file: 'dist/esm.js',
-		format: 'esm',
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel(),
-		banner(genHeader('data'))
-	],
-},
-{
-	input: 'index.js',
-	output: {
-		file: 'dist/umd.js',
-		format: 'umd',
-		exports: 'named',
-		name: 'vis',
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel(),
-		banner(genHeader('data'))
-	],
-}]
+// TypeScript because Babel transpiles modules in isolation, therefore no type reexports.
+// CommonJS because Babel is not 100 % ESM.
+
+const babelConfingBase = {
+  extensions: ['.mts', '.ts', '.tsx', '.mjs', '.js', '.jsx'],
```

Comment:
I don't think we need all these extensions. I think it is more confusing than helpful.